src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -215,4 +247,39 @@ class EthService(blockchain: Blockchain, blockGenerator: BlockGenerator, appStat
     }
   }
 
+  def call(req: CallRequest): ServiceResponse[CallResponse] = {
+    val fromAddress = req.tx.from
+      .map(Address.apply) // `from` param, if specified
+      .getOrElse(
+        keyStore
+          .listAccounts().getOrElse(Nil).headOption // first account, if exists and `from` param not specified
+          .getOrElse(Address(0))) // 0x0 default
+
+    val toAddress = req.tx.to.map(Address.apply)
+    val tx = Transaction(0, req.tx.gasPrice, req.tx.gas, toAddress, req.tx.value, req.tx.data)
+    val stx = SignedTransaction(tx, ECDSASignature(0, 0, 0.toByte), fromAddress)
+
+    Future.successful {
+      resolveBlock(req.block).map { block =>
+        val txResult = ledger.simulateTransaction(stx, block.header, blockchainStorages, validators)
+        CallResponse(txResult.vmReturnData)
+      }
+    }
+  }
+
+  private def resolveBlock(blockParam: BlockParam): Either[JsonRpcError, Block] = {
+    def getBlock(number: BigInt): Either[JsonRpcError, Block] = {
+      blockchain.getBlockByNumber(number)
+        .map(Right.apply)
+        .getOrElse(Left(JsonRpcErrors.InvalidParams(s"Block $number not found")))
+    }
+
+    blockParam match {
+      case BlockParam.WithNumber(blockNumber) => getBlock(blockNumber)
+      case BlockParam.Earliest => getBlock(0)
+      case BlockParam.Latest => getBlock(appStateStorage.getBestBlockNumber())
+      case BlockParam.Pending => getBlock(appStateStorage.getBestBlockNumber())
```

Comment:
The pending block should be obtained from the pending txs, righ? Should we add a comment and create a task for obtaining it?